torch/csrc/jit/tensorexpr/stmt.h

```diff
@@ -78,16 +90,37 @@ class Block : public StmtNode<Block> {
     }
     return new Block(valid_stmts);
   }
+
   int nstmts() const {
     return stmts_.size();
   }
-  Stmt* stmt(int index) const {
-    return stmts_[index];
+
+  void append_stmt(Stmt *s) {
+    stmts_.push_back(s);
```

Comment:
Check that s->get_parent() == nullptr? Otherwise, we need to make sure the old parents are no longer referring to "s".